nginx/lua/redirect_content.lua

```diff
@@ -0,0 +1,96 @@
+local ngx = require("ngx")
+local resty_http = require("resty.http")
+local fileutil = require("fileutil")
+local config = require("config")
+local gossip = require("gossip")
+
+
+-- TODO: always "redirect"?
+local path = ngx.var.request_uri:sub(#"/redirect" + 1)
+local webdav_uri = config.data.uriprefix .. path
+local stat = fileutil.get_metadata(config.data.local_path .. path, false)
+if stat.exists then
+    ngx.status = ngx.HTTP_TEMPORARY_REDIRECT
+    -- TODO: do we want to use the full url (config.data.server_address)
+    ngx.header["Location"] = webdav_uri
```

Comment:
If the current server has the file we would rather issue an internal rewrite since we can respond directly to the client rather than make them have to issue a new request. This means this redirect code would better be in a different request stage.